lib/shimmer.dart

```diff
@@ -43,19 +43,25 @@ class Shimmer extends StatefulWidget {
               0.65,
               1.0
             ]),
-        super(key: key);
+        super(key: key) {
```

Comment:
Why don't use `assert(period.inMilliseconds != null && period.inMilliseconds < Duration(milliseconds: 0).inMilliseconds)`?
By the way, you can simply throw a String instead of an exception object.